cli/azd/internal/repository/app_init.go

```diff
@@ -0,0 +1,236 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/internal/appdetect"
+	"github.com/azure/azure-dev/cli/azd/internal/scaffold"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/output/ux"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/otiai10/copy"
+)
+
+var languageMap = map[appdetect.Language]project.ServiceLanguageKind{
+	appdetect.DotNet:     project.ServiceLanguageDotNet,
+	appdetect.Java:       project.ServiceLanguageJava,
+	appdetect.JavaScript: project.ServiceLanguageJavaScript,
+	appdetect.TypeScript: project.ServiceLanguageTypeScript,
+	appdetect.Python:     project.ServiceLanguagePython,
+}
+
+var dbMap = map[appdetect.DatabaseDep]struct{}{
+	appdetect.DbMongo:    {},
+	appdetect.DbPostgres: {},
+}
+
+var ErrNoServicesDetected = errors.New("no services detected")
+
+// InitFromApp initializes the infra directory and project file from the current existing app.
+func (i *Initializer) InitFromApp(
+	ctx context.Context,
+	azdCtx *azdcontext.AzdContext,
+	initializeEnv func() error) error {
+	i.console.Message(ctx, "")
+	title := "Scanning app code in current directory"
+	i.console.ShowSpinner(ctx, title, input.Step)
+	wd := azdCtx.ProjectDirectory()
+
+	projects := []appdetect.Project{}
+	start := time.Now()
+	sourceDir := filepath.Join(wd, "src")
+	// Prioritize src directory if it exists
+	if ent, err := os.Stat(sourceDir); err == nil && ent.IsDir() {
+		prj, err := appdetect.Detect(sourceDir)
```

Comment:
Guessing that not including the exclude patterns for eng, tool and tools as you do later was an explicit choice here (since they are less likely to be found under a folder named `src`?) but calling it out in case you intended to do this both places.

Reply:
Indeed, that's how it's written for now. We can finetune these heuristics over time.